GitUI/UserControls/ToolStripEx.cs

```diff
@@ -18,6 +19,64 @@ public ToolStripEx()
             _gripButton = propGrip.GetValue(this) as ToolStripButton;
         }
 
+        protected override void OnItemAdded(ToolStripItemEventArgs e)
+        {
+            base.OnItemAdded(e);
+
+            if (e.Item is ToolStripDropDownItem item)
+            {
+                item.DropDownOpening += SplitButton_DropDownOpening;
+                item.DropDownClosed += SplitButton_DropDownClosed;
+            }
+        }
+
+        protected override void OnItemRemoved(ToolStripItemEventArgs e)
+        {
+            if (e.Item is ToolStripDropDownItem item)
+            {
+                item.DropDownOpening -= SplitButton_DropDownOpening;
+                item.DropDownClosed -= SplitButton_DropDownClosed;
+            }
+        }
+
+        [DllImport("user32.dll")]
+        private static extern int SendMessage(IntPtr hwnd, int wmsg, bool wparam, int lparam);
+
+        private const int WM_SETREDRAW = 11;
+
+        private static void SuspendDrawing(Control control)
+        {
+            if (control != null)
+            {
+                SendMessage(control.Handle, WM_SETREDRAW, false, 0);
+            }
+        }
+
+        private static void ResumeDrawing(Control control)
+        {
+            if (control != null)
+            {
+                SendMessage(control.Handle, WM_SETREDRAW, true, 0);
+                control.Refresh();
+            }
+        }
```

Comment:
These two methods have single call site each, and should be inlined. There's no need to create more stack frames.

Reply:
I understand you point of view, but since source code is for humans and not computers, I since it is better to have short and well named methods than saving a stack frame.

Let me know if this is a blocker and I will change as you requested, but I'd like to make sure before.

Reply:
> but since source code is for humans and not for computers,

There are many different schools of thought and opinions on the subject, but small methods which are only called from one place impose maintenance and mental tax too. (Not to mention potential have perf implications).

For the sake of discussion, suppose, I follow your argument, let's compare the proposed code:
```cs
        private static void SuspendDrawing(Control control)
        {
            if (control != null)
            {
                NativeMethods.SendMessageW(control.Handle, NativeMethods.WM_SETREDRAW, NativeMethods.FALSE, IntPtr.Zero);
            }
        }

        private void SplitButton_DropDownOpening(object? sender, EventArgs e)
        {
            if (sender is ToolStripDropDownItem item)
            {
                SuspendDrawing(item.Owner);
            }
        }
```
...and an alternative:
```cs
        private void SplitButton_DropDownOpening(object? sender, EventArgs e)
        {
            if (sender is ToolStripDropDownItem item && item.Owner is Control control)
            {
                NativeMethods.SendMessageW(control.Handle, NativeMethods.WM_SETREDRAW, NativeMethods.FALSE, IntPtr.Zero);
            }
        }
```

Could you please articulate why the first snippet is better? It's more verbose, it has multiple `if` conditions and, thus, more complex.

Reply:
It is more verbose if you take it in its whole, but let's just take the entry points:

```cs
        private void SplitButton_DropDownOpening(object? sender, EventArgs e)
        {
            if (sender is ToolStripDropDownItem item)
            {
                SuspendDrawing(item.Owner);
            }
        }
```

It reads like a book, reading this I know that entering the `if`, it will suspend redraws, and most of the time that's all I need to know. If I choose or need to know how redraws are suspended, I can decide to visit the `SuspendDrawing()` methods. I can also decide to not spend time for that and move on.

With the follwoing:

```cs
        private void SplitButton_DropDownOpening(object? sender, EventArgs e)
        {
            if (sender is ToolStripDropDownItem item && item.Owner is Control control)
            {
                NativeMethods.SendMessageW(control.Handle, NativeMethods.WM_SETREDRAW, NativeMethods.FALSE, IntPtr.Zero);
            }
        }
```

First, the `if` condition is longer and more complex, so it's harder, when overlooking the code, to understand how and when the `if` will enter. Second, the statement in the `if` barely tells what it does by itself, and one will probably have to search MSDN or else to understand what's going on. You could add a comment, but then that will make the code longer and therefore more verbose. Also, comment may go desync the day one decide to send another type of message for some reason but forget to update the comment. You may argue that it's also possible to change the message and forgetting to rename the method, but it's less likely to happen when the function does just one single thing. I believe code beats comments, so I prefer to have a short and well named methods that's self-describing, than obscure code commented somewhere. One more thing, the day someone else needs that feature too, the code is already factorized (partly) and ready to use. With the second snippet, it's easy to move the content of the `if` and forget about also taking the `if` for null check. With the first snippet, moving the method to a dedicated class (could be moved to `NativeMethods`) is all that's to be done as it is already self-sufficient and self-describing.

As for performance, it is also a very important topic for me and I agree with you, but here we are not in a hot path at all, so trying to save a stack frame is not really something that should drive this decision, IMHO.

Reply:
Thank you for the explanation.

Reply:
I inlined the methods.